### PR TITLE
Implemented dropdown menu generator

### PR DESCRIPTION
**Pull request by djmiley**
Part of #259

### CODE REVIEW

**Comment by djmiley** [comment]
This seems lke an opportunity to refactor some of the data stuff. The valueString and option are both set to the same thing in the data options, so there's an oppurtunity for some extra information being held there. Maybe a new dataType object containing information on whether to display the period change dropdown?

File: src/assets/js/menu/generator/dropdownGroup.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+(function(d3, fc, sc) {
+    'use strict';
+
+    sc.menu.generator.dropdownGroup = function() {
+        var dispatch = d3.dispatch('optionChange');
+
+        function layoutDropdown(sel) {
+            sel.selectAll('option')
```
Comment (jleft) [inline src/assets/js/menu/generator/dropdownGroup.js]: Could use dataJoin?

Comment (djmiley) [inline src/assets/js/menu/generator/dropdownGroup.js]: I wasn't able to get this to work

Comment (djmiley) [inline src/assets/js/menu/generator/dropdownGroup.js]: Silly error

File: src/index.html [hunk 2]
```diff
@@ -22,15 +22,8 @@
...
-            <select id="type-selection" selected="type" class="form-control">
-              <option value="generated">Data generator</option>
-              <option value="bitcoin">Bitcoin data</option>
-            </select>
-            <select id="period-selection" selected="period" class="form-control">
-              <option value="3600">1 hr</option>
-              <option value="300">5 min</option>
-              <option value="60">1 min</option>
-            </select>
+            <select id="type-dropdown" selected="type" class="form-control"></select>
```
Comment (tlsim) [inline src/index.html]: I think we have lost some styling here - see menu.less

File: src/assets/js/menu/data/type.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+(function(d3, fc, sc) {
+    'use strict';
+
+    sc.menu.data.type = function() {
```
Comment (tlsim) [inline src/assets/js/menu/data/type.js]: 'product' or 'instrument' might be better. 'type' brings this to mind: https://en.wikipedia.org/wiki/Class_%28computer_programming%29#Class_vs._type

Comment (djmiley) [inline src/assets/js/menu/data/type.js]: i'll go with product?

File: src/assets/js/menu/data/period.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        var hourPeriod = sc.menu.option('1 hr', '3600', 3600);
+        var fiveMinutePeriod = sc.menu.option('5 min', '300', 300);
+        var oneMinutePeriod = sc.menu.option('1 min', '60', 60);
+
+        var options = sc.menu.generator.dropdownGroup()
+            .on('optionChange', function(period) {
+                dispatch.dataPeriodChange(period);
+            });
+
+        var dataPeriodChangeMenu = function(selection) {
```
Comment (tlsim) [inline src/assets/js/menu/data/period.js]: This pattern (binding a collection of *options* to a selection, calling a *generator* on the selection and forwarding an option change event) is repeated in a few places now. Can you abstract this idea? (not for this PR) #260

Comment (djmiley) [inline src/assets/js/menu/data/period.js]: The below commit arguably displays this in action

**Pull request merged**